### PR TITLE
feat: add all scripts from multisig to governance

### PR DESCRIPTION
**Pull request by 0xtekgrinder**

### CODE REVIEW

File: scripts/Constants.s.sol [hunk 1]
```diff
@@ -31,11 +67,35 @@ address constant SAFE_MAINNET = 0xdC4e6DFe07EFCa50a197DF15D9200883eF4Eb1c8;
...
 // TODO update when deployed
 // address constant proposalSender = address(0x499C86959a330Eb860FdFFf6e87896d4298a4F4E);
 address constant proposalReceiverPolygon = address(0x060246eD061999F7e128Fd8355d84467d6726b71);
 address constant timelockPolygon = address(0x0d17B69fF7D30F7EC13A9447d1E5624b601a730b);
+
+/*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                  EXTERNAL CONTRACTS                                                
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+address constant EUROC = 0x1aBaEA1f7C830bD89Acc67eC4af516284b1bC33c;
```
Comment (GuillaumeNervoXS) [inline scripts/Constants.s.sol:93]: I don't think we need this here

File: scripts/Interfaces.s.sol [hunk 2]
```diff
@@ -0,0 +1,59 @@
+pragma solidity ^0.8.19;
+
```
Comment (GuillaumeNervoXS) [inline scripts/Interfaces.s.sol:2]: I encountered that in the angle-multisig, maybe better to create interfaces in respective repo to not copy paste evrywhere

File: scripts/Utils.s.sol [hunk 3]
```diff
@@ -109,4 +51,237 @@ contract Utils is Script {
         if (res.length == 0) revert("Chain not supported");
         return uint16(stringToUint(string(res)));
     }
+
+    enum ContractType {
```
Comment (GuillaumeNervoXS) [inline scripts/Utils.s.sol:55]: You can import from `Constants.s.sol`

File: scripts/Utils.s.sol [hunk 4]
```diff
@@ -109,4 +51,237 @@ contract Utils is Script {
...
+        else revert("contract not supported");
+
+        bytes memory res = vm.ffi(cmd);
+        // When process exit code is 1, it will return an empty bytes "0x"
+        if (res.length == 0) revert("Chain not supported");
+        console.log(address(bytes20(res)));
+        return address(bytes20(res));
+    }
+
+    function wrapTimelock(
```
Comment (GuillaumeNervoXS) [inline scripts/Utils.s.sol:109]: All below maybe better to isolate in a specific contract

File: scripts/governance/RevokeMultiSig.s.sol [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+import { Utils } from "../Utils.s.sol";
+import "../Constants.s.sol";
+
+interface ITreasuryWithRole {
+    function addMinter(address minter) external;
+
+    function removeMinter(address minter) external;
+}
+
+contract RevokeMultiSig is Utils {
```
Comment (GuillaumeNervoXS) [inline scripts/governance/RevokeMultiSig.s.sol:19]: This exact one is useless as it is specific to introduce the on chain governance

**Review by GuillaumeNervoXS — COMMENTED** [review]
I prefer to have only a few script that works and then adapt to all the ones in the multisig

File: helpers/createProposal.sh [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+    echo "Running test"
+    FOUNDRY_PROFILE=dev forge test --match-contract $testContract -vvv
+
+    echo ""
+    echo "Would you like to create the proposal ? (yes/no)"
+    read execute
+
+    # if [[ $execute == "yes" ]]; then
+    #     forge script scripts/interaction/Propose.s.sol:Propose --fork-url $mainnet_uri --broadcast
+    # fi
```
Comment (0xtekgrinder) [inline helpers/createProposal.sh:194]: don't forget to uncomment this

File: test/unit/SimulationSetup.t.sol [hunk 7]
```diff
@@ -204,6 +204,7 @@ contract SimulationSetup is Test {
                                                         HELPERS                                                     
     //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
 
+    // TODO USE IT
```
Comment (0xtekgrinder) [inline test/unit/SimulationSetup.t.sol:207]: remove this forgotten comment

**Pull request merged**